tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,345 @@
+
+<!-- What is the positive outcome for the customer if we deliver these features, i.e. what is the value proposition for the customer? Remember, this is customer-centric. -->
+
+## Key scenarios
+<!-- List ~3-7 high level scenarios to clarify the value of the Epic and point to how we will decompose this big area into component capabilities. We may ultimately have more than one level of scenario. -->
+
+### Scenario 1: As an operator I can install and configure Radius with my GitOps tool of choice
+<!-- One or two sentence summary -->
+My team uses GitOps to manage Kubernetes clusters and applications that run in those clusters.  All relevant configuration information is stored in a git repository that serves as the source of truth for both the application code and infrastructure configuration.  To use Radius, I need to include Radius as a dependency for my Kubernetes cluster, just as I would any other dependency, via my GitOps tools.
+
+### Scenario 2: As an operator or developer, I can deploy and manage Radius Environments, Recipes, applications, and resources via my GitOps toolset
+<!-- One or two sentence summary -->
+With Radius installed, as an Operator, I now define, deploy and manage a Radius environment and associate relevant Recipes to that environment. 
+
+As a Developer, my Operations counterpart has provided a Radius Environment plus Recipes to enable my Radius application development.  I have familiarized myself with Radius and have experimented with deploying my application to this environment using the Recipe.  After my app is deployed, I can view its graph in the Radius dashboard.  Everything is working well for me so far and I really appreciate these cool Radius features! 
+
+### Scenario 3: As a Site Reliability Engineer (SRE), I can patch and edit the configuration in my Kubernetes cluster through my GitOps toolset.
+<!-- One or two sentence summary -->
+As an SRE, my Operations counterpart has provided a Kubernetes cluster, a Radius Environment running on that cluster and the infrastructure resources through Radius Recipes required for the Radius Applications that will run in that Radius Environment. I am expected to make adjustments to the configuration of the Kubernetes cluster and the Radius Environment as needed. I can do this through my GitOps toolset.
+
+### Stretch Goal: As an Operator and as a Developer, I can visualize the changes tracked by my GitOps toolset on my Radius Dashboard
+We still need to flesh this out but the basic idea is the Radius dashboard can leverage the Notification Controller in Flux and ArgoCD to display to users inbound and outbound information of the source changes and cluster changes affecting their application and resources.
```

Comment:
Have we done any research here? Are there existing plugins we can leverage?

Reply:
Not yet, but since this is beyond scope of the initial feature implementation i'm moving this to out of scope.